component/src/action_handler/transaction/stateful.rs

```diff
@@ -32,3 +33,31 @@ pub(super) async fn fmd_parameters_valid<S: StateRead>(
         height,
     )
 }
+
+pub(super) async fn timestamps_are_valid<S: StateRead>(
```

Comment:
I wasn't sure exactly where to add the checks for these fields, and what the difference between a panic and an `Err()` is in this context, this is my best guess. I also couldn't find any check for the expiry height anywhere, so I added it here.